src/services/postService.js

```diff
@@ -119,6 +125,43 @@ async function checkLike(like, postID, userID){
     return postData;
 }
 
+async function checkTags(tags, inclusive){
+    const posts = await postDAO.scanPosts();
+    throwIfError(posts);
+    if (!tags){
+        return posts.Items;
+    }
+    const postSet = new Set();
+    if (inclusive == 1){
+        for (const post of posts.Items){
+            for (const i of tags){
+                if (!post.tags){
+                    break;
+                }
+                if (post.tags.has(i)){
+                    postSet.add(post);
+                    break;
+                }
+            }
+        }
+    }
+    else {
+        for (const post of posts.Items){
+            let should = true;
+            for (const i of tags){
+                if (!post.tags || !post.tags.has(i)){
+                    should = false;
+                    break;
+                }
+            }
+            if (should){
+                postSet.add(post);
+            }
+        }
+    }
+    return [...postSet];
```

Comment:
It would remove at least some need but that filter expression does not account for the fact that the user could search multiple tags and ask for posts that include both tags only or posts that have either tag.